blockreassurance.php

```diff
@@ -245,8 +245,15 @@ public function loadAsset()
      */
     private function folderUploadFilesHasGoodRights()
     {
-        return is_writable($this->folder_file_upload)
-            && is_executable($this->folder_file_upload);
+        if (is_writable($this->folder_file_upload)) {
+            if (preg_match('/^[a-zA-Z]{1}\:{1}\\\\{1}/', $this->folder_file_upload) !== 1) {    // do not check is_executable on windows platform - always false
+                return is_executable($this->folder_file_upload);
+            }
+
+            return true;
+        }
+
+        return false;
```

Comment:
I think it can be simplified with:
```suggestion
        // do not check is_executable on windows platform (https://www.php.net/manual/en/function.is-executable.php#refsect1-function.is-executable-notes)
        return is_writable($this->folder_file_upload) && !(preg_match('/^[a-zA-Z]{1}\:{1}\\\\{1}/', $this->folder_file_upload) !== 1) || is_executable($this->folder_file_upload;
```

What do you think?